backend/src/admin/log/repository/log.repository.ts

```diff
@@ -0,0 +1,105 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { LogPagenationDto } from 'src/admin/dto/log.pagenation.dto';
+import { IAdminLogRepository } from 'src/admin/log/repository/log.interface.repository';
+import Cabinet from 'src/entities/cabinet.entity';
+import LentLog from 'src/entities/lent.log.entity';
+import { Repository } from 'typeorm';
+
+export class AdminLogRepository implements IAdminLogRepository {
+    constructor(
+      @InjectRepository(LentLog)
+      private LogRepository: Repository<LentLog>,
+    ) {}
+
+async getLentLogByUserId(
+    userId: number,
+    index: number,
+    length: number,
+  ): Promise<LogPagenationDto> {
+    // lent_log 테이블과 cabinet 테이블을 조인함.
+    const result = await this.LogRepository
+      .createQueryBuilder('ll')
+      .select([
+        'll.log_user_id',
+        'll.log_intra_id',
+        'll.log_cabinet_id',
+        'll.lent_time',
+        'll.return_time',
+        'COUNT(*) OVER () AS cnt',
+      ])
+      .leftJoin(Cabinet, 'c', 'll.log_cabinet_id = c.cabinet_id')
+      .addSelect([
+        'c.cabinet_id',
+        'c.cabinet_num',
+        'c.location',
+        'c.section',
+        'c.floor',
+      ])
+      .where('ll.log_user_id = :user_id', { user_id: userId })
+      .limit(length)
+      .offset(index)
```

Comment:
1. `offset` 대신 `skip` 함수를 이용하는게 좋을 것 같습니다.
> Set's OFFSET - selection offset. NOTE that it may not work as you expect if you are using joins. If you want to implement pagination, and you are having join in your query, then use instead skip method instead.
2. index를 page로 변경
은비킴님이랑만 먼저 얘기를 했었는데 페이지네이션에서 클라이언트가 index 대신 가져올 page를 건네주는게 사용하는 입장에서 더 의미를 파악하기 좋을 것 같습니다..!
최종적으로 해당 라인은 `skip: page * length,`로 수정하는걸 제안합니다 😊

Reply:
아직 페이지네이션에 대한 개념이 부족해서 가능하시다면 직접 해주시면 감사하겠습니다..! (나중에 알려주세요..)

Reply:
알겠습니다~